src/material.h

```diff
@@ -88,4 +115,51 @@ class Metal : public Material
 	}
 };
 
+
+class Dielectric : public Material
+{
+public:
```

Comment:
I've seen that in all the material classes, all the attribute are defined as `public`?
Would't be better to make them `private`?